crates/sui-indexer/src/handlers/checkpoint_handler_v2.rs

```diff
@@ -694,11 +695,12 @@ fn try_create_dynamic_field_info(
             let version = object.version();
             let digest = object.digest();
             let object_type = object.data.type_().unwrap().clone();
+            let struct_tag: StructTag = object_type.into();
             DynamicFieldInfo {
                 name,
                 bcs_name,
                 type_,
-                object_type: object_type.to_string(),
+                object_type: struct_tag.to_canonical_string_with_prefix(),
```

Comment:
this is part of the writing path, adding the prefix here will require a backfill from genesis for data consistency.
is there a way to keep the object type & df_object_type as it is in the DB and only adding the prefix on the reading path?

Reply:
Either way, what we have in the db is wrong since this uses short str